src/wallet/api.rs

```diff
@@ -644,48 +662,31 @@ impl Wallet {
         Ok(all_utxos)
     }
 
-    pub(crate) fn get_all_locked_utxo(&self) -> Result<Vec<ListUnspentResultEntry>, WalletError> {
-        let all_utxos = self
-            .rpc
-            .list_unspent(Some(0), Some(9999999), None, None, None)?;
-        Ok(all_utxos)
-    }
     /// Returns a list all utxos with their spend info tracked by the wallet.
     /// Optionally takes in an Utxo list to reduce RPC calls. If None is given, the
     /// full list of utxo is fetched from core rpc.
     pub fn list_all_utxo_spend_info(
         &self,
-        utxos: Option<&Vec<ListUnspentResultEntry>>,
     ) -> Result<Vec<(ListUnspentResultEntry, UTXOSpendInfo)>, WalletError> {
-        let all_utxos = if let Some(utxos) = utxos {
-            utxos.clone()
-        } else {
-            self.get_all_utxo()?
-        };
-
-        let processed_utxos = all_utxos
-            .iter()
-            .filter_map(|utxo| {
-                let mut spend_info = self.check_if_fidelity(utxo);
-                if spend_info.is_none() {
-                    spend_info = self.check_if_live_contract(utxo).unwrap();
-                }
-                if spend_info.is_none() {
-                    spend_info = self.check_descriptor_utxo_or_swap_coin(utxo).unwrap();
-                }
-                spend_info.map(|info| (utxo.clone(), info))
+        let processed_utxos = self
+            .store
+            .utxo_cache
+            .values()
+            .filter_map(|(utxo, _)| {
+                self.check_if_fidelity(utxo)
+                    .or_else(|| self.check_if_live_contract(utxo).unwrap())
+                    .or_else(|| self.check_descriptor_utxo_or_swap_coin(utxo).unwrap())
+                    .map(|info| (utxo.clone(), info))
```

Comment:
Ugh... you are still processing!! each of these functions returns the `UtxoSpendInfo` through costly searches. You already have the  spend info inside the map for each utxos. Why recreate them again?

Reply:
We already did the process loop (exact same filter in the update function)  while updating the internal map. Doing it again here is redundant. 

The whole point of having the spend info in the cache is to not call these functions repetitively.

Reply:
We should rename these functions for better clarity. its not just checking its also returning the spend info. So something like `derive_live_contract_spend_info()` with a doc on the top that `these are costly searches and should be used with care.`